server/src/index.js

```diff
@@ -106,52 +121,52 @@ const protocol = (process.env.PROTOCOL || 'http') === 'http' ? http : https;
 const server = protocol.createServer(app.callback());
 const io = Io(server, {
   pingInterval: 20000,
-  pingTimeout: 5000
+  pingTimeout: 5000,
 });
-io.adapter(socketRedis(process.env.REDIS_URL));
+
+// Only use redis if we have a REDIS_URL in config
+if (process.env.REDIS_URL !== undefined) {
+  io.adapter(socketRedis(process.env.REDIS_URL));
```

Comment:
Maybe we initializer `io` before creating a `store`, then we can do this portion on the store setup

Reply:
After looking below, we'd want our store to know that it's using a custom io adapter so we'd want to make that flag readily available for use, like setting a `store.hasAdapter = bool` flag after initializing the adapter.

Reply:
I've added the boolean `hasSocketAdapter` property and a `getSocketAdapter` to creat the adapter on demand.